src/monitor-specs.js

```diff
@@ -1,36 +1,262 @@
-import monitorList from "./data/monitor.json" with { type: "json" };
-
-const toGhUrl = repo => `https://${repo.split("/")[0].toLowerCase()}.github.io/${repo.split("/")[1]}/`;
-
-const today = new Date().toJSON().slice(0, 10);
-
-// Check last-modified HTTP header for specs to highlight those that needs
-// re-review
-let review_needed = [];
-const candidates = Object.keys(monitorList.repos).map(r => {return {...monitorList.repos[r], url: toGhUrl(r)};}).concat(
-  Object.keys(monitorList.specs).map(s => {return {...monitorList.specs[s], url: s};}));
-for (let candidate of candidates) {
-  const response = await fetch(candidate.url);
-  const { headers } = response;
-  // The CSS drafts use a proprietary header to expose the real last modification date
-  const lastRevised = headers.get('Last-Revised') ? new Date(headers.get('Last-Revised')) : new Date(headers.get('Last-Modified'));
-  if (lastRevised > new Date(candidate.lastreviewed)) {
-    review_needed.push({...candidate, lastupdated: lastRevised.toJSON()});
-  }
-  // We don't need the response's body, but not reading it means Node will keep
-  // the network request in memory, which prevents the CLI from returning until
-  // a timeout occurs.
-  await response.arrayBuffer();
+
+/**
+ * The monitor-specs script loops through the list of open issues in the
+ * browser-specs repository that have a "new spec" label, checks those that
+ * have not been reviewed for a while, and adds a comment and "review" label to
+ * those that seems worth reviewing again because an update was detected since
+ * last review.
+ *
+ * The last time that an issue was reviewed is the last time that the "review"
+ * label was removed, which the script retrieves thanks through the GraphQL
+ * endpoint.
+ *
+ * To report the list of issues that need a review (without updating the
+ * issues), run:
+ * node src/monitor-specs.js
+ *
+ * To report the list of issues that need a review and updated the issues to
+ * add a comment/label to the issues, run:
+ * node src/monitor-specs.js --update
+ */
+
+import sendGraphQLQuery from "./graphql.js";
+import splitIssueBodyIntoSections from "./split-issue-body.js";
+import loadJSON from "./load-json.js";
+
+const config = await loadJSON("config.json");
+const githubToken = config?.GITHUB_TOKEN ?? process.env.GITHUB_TOKEN;
+
+
+/**
+ * The list of specs that are already known is derived from open and closed
+ * issues in the browser-specs repository.
+ */
+const BROWSER_SPECS_REPO = {
+  owner: "w3c",
+  name: "browser-specs"
+};
+
+
+/**
+ * Script does not update GitHub issues by default
+ */
+const updateGitHubIssues =
+  (process.argv[2] === "--update") ||
+  (process.argv[2] === "-u");
+
+
+/**
+ * Retrieve the list of specs and repositories that should not be reported
+ * because we're already aware of them and their treatment is still pending or
+ * we explicitly don't want to add them to browser-specs.
+ */
+async function fetchIssuesToReview() {
```

Comment:
the logic for paging through graphql results could be factored out in graphql.js?

Reply:
The insertion points for `after` and `pageInfo` depend on the query at hand. Automatic handling would require parsing the query string and going through the response's structure. Theoretically, paging could also be needed at different levels.

I could create a higher level function that's specific to returning a list of issues that match certain criteria, but I'd say that with 2+1 relatively simple queries, we haven't yet crossed the line that makes that factoring out worth the hassle. We could end up re-writing some of the queries to use another approach. For example, issue queries could be re-written as a GraphQL query for a label with issues appearing at a nested level.